pkg/autodiscovery/providers/utils.go

```diff
@@ -31,34 +31,56 @@ func init() {
 	config.Datadog.SetDefault("autoconf_template_url_timeout", 5)
 }
 
-// parseJSONValue returns a slice of ConfigData parsed from the JSON
+// parseJSONValue returns a slice of slice of ConfigData parsed from the JSON
 // contained in the `value` parameter
-func parseJSONValue(value string) ([]integration.Data, error) {
+func parseJSONValue(value string) ([][]integration.Data, error) {
 	if value == "" {
 		return nil, fmt.Errorf("Value is empty")
 	}
 
 	var rawRes []interface{}
-	var result []integration.Data
+	var result [][]integration.Data
 
 	err := json.Unmarshal([]byte(value), &rawRes)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to unmarshal JSON: %s", err)
+		return nil, fmt.Errorf("failed to unmarshal JSON: %s", err)
 	}
 
 	for _, r := range rawRes {
 		switch r.(type) {
-		case map[string]interface{}:
-			init, _ := json.Marshal(r)
-			result = append(result, init)
+		case []interface{}:
+			objs := r.([]interface{})
+			var subResult []integration.Data
```

Comment:
If never an error append during the loop, the result is not modified. 
But it is an extra precaution since we return `result=nil` in case if error...